Add MT percentage filtering

tbrittoborges opened the pull request:
This PR adds support to mitochondrial fraction filtering per sample, a widely used quality metric in snRNAseq.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scdownstream/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/scdownstream _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

tbrittoborges committed:
Add mito_fraction from sample input

tbrittoborges commented:
Closes #87

nictru commented:
Hey, a couple of small things:

1. Make sure there are no conflicts with the `dev` branch (`rebase`/`merge`)
2. As far as I can tell, scanpy gives you a percentage (0-100) while the new pipeline parameter is a fraction (0-1). I might be wrong, did not test. I would prefer to change the pipeline parameter to a percentage in this case (integer between 0 and 100).
3. Please add a `max_` prefix to the column name in the samplesheet. I know only the maximum makes sense here, but for consistency with other thresholds I think it makes sense.

tbrittoborges committed:
Add support to celltypist model.convert via `celltypist_map_file` parameter.
Fix #98

tbrittoborges commented:
>     1. Make sure there are no conflicts with the `dev` branch (`rebase`/`merge`)
Resolved the merge conflicts on `assets/schema_input.json` and  `docs/usage.md`.

>     2. As far as I can tell, scanpy gives you a percentage (0-100) while the new pipeline parameter is a fraction (0-1). I might be wrong, did not test. I would prefer to change the pipeline parameter to a percentage in this case (integer between 0 and 100).
Changed from float to int.

>     3. Please add a `max_` prefix to the column name in the samplesheet. I know only the maximum makes sense here, but for consistency with other thresholds I think it makes sense.
Added the `max` prefix.

tbrittoborges committed:
Merge branch 'dev' into mitochondrial_fraction

tbrittoborges committed:
Max mitochondrial fraction. Fix #97

tbrittoborges committed:
prettier usage.md

tbrittoborges committed:
rm celltypist_map_file changes

tbrittoborges committed:
lint modules.json

tbrittoborges committed:
prettier modules.json

nictru commented on docs/usage.md:
```diff
@@ -107,6 +108,10 @@ genome: 'GRCh37'
 
+### Cell type annotation
+
+Automated cell type annotation using [Celltypist](https://github.com/Teichlab/celltypist) is supported. You can specify the models to use with the `celltypist_model` parameter. If no models are specified, no cell type annotation will be performed.
```
```suggestion
Automated cell type annotation using [Celltypist](https://github.com/Teichlab/celltypist) is supported. You can specify the models to use with the [`celltypist_model` parameter](https://nf-co.re/scdownstream/dev/parameters/#celltypist_model). If no models are specified, no cell type annotation will be performed.
```

nictru committed:
Add celltypist model parameter link
